### PR TITLE
Disable `accessibilityState` when `TouchableWithoutFeedback` is `disabled`

### PR DESCRIPTION
**Pull request by carloscuesta**
## Summary

Disable `accessibilityState` when the `TouchableWithoutFeedback` is `disabled`. This fixes https://github.com/facebook/react-native/issues/30953

## Changelog

[General] [Changed] - Disable `accessibilityState` when the `TouchableWithoutFeedback` is `disabled`.

## Test Plan

Tested the `TouchableWithoutFeedback` component on an Android device

### CODE REVIEW

File: Libraries/Components/Touchable/__tests__/__snapshots__/TouchableWithoutFeedback-test.js.snap [hunk 1]
```diff
@@ -15,3 +15,94 @@ exports[`TouchableWithoutFeedback renders correctly 1`] = `
...
+  onResponderRelease={[Function]}
+  onResponderTerminate={[Function]}
+  onResponderTerminationRequest={[Function]}
+  onStartShouldSetResponder={[Function]}
+/>
+`;
+
+exports[`TouchableWithoutFeedback with disabled state should be disabled when disabled is true and accessibilityState is empty 1`] = `
+<View
+  accessibilityState={Object {}}
```
Comment (kacieb) [inline Libraries/Components/Touchable/__tests__/__snapshots__/TouchableWithoutFeedback-test.js.snap:40]: It looks like there's a couple of cases where accessibilityState is not including `disabled: true` when it should (including this one) - perhaps you need to re-run the snapshot tests or investigate in the code why this is the case?

File: Libraries/Components/Touchable/__tests__/__snapshots__/TouchableWithoutFeedback-test.js.snap [hunk 2]
```diff
@@ -15,3 +15,94 @@ exports[`TouchableWithoutFeedback renders correctly 1`] = `
...
+  onResponderTerminate={[Function]}
+  onResponderTerminationRequest={[Function]}
+  onStartShouldSetResponder={[Function]}
+/>
+`;
+
+exports[`TouchableWithoutFeedback with disabled state should keep accessibilityState when disabled is true 1`] = `
+<View
+  accessibilityState={
+    Object {
```
Comment (kacieb) [inline Libraries/Components/Touchable/__tests__/__snapshots__/TouchableWithoutFeedback-test.js.snap:75]: Here's another where disabled is missing.

**Review by kacieb — CHANGES_REQUESTED** [review]

File: Libraries/Components/Touchable/__tests__/__snapshots__/TouchableWithoutFeedback-test.js.snap (hunk 1 above)
Comment (carloscuesta) [reply Libraries/Components/Touchable/__tests__/__snapshots__/TouchableWithoutFeedback-test.js.snap:40]: Thanks for the review, I'm going to take a look at it!

Comment (carloscuesta) [reply Libraries/Components/Touchable/__tests__/__snapshots__/TouchableWithoutFeedback-test.js.snap:40]: Found the issue, it's related with this code of TouchableWithoutFeedback:

https://github.com/facebook/react-native/blob/08ef1df1bad8408a75a195dda698db184efea0b7/Libraries/Components/Touchable/TouchableWithoutFeedback.js#L122-L126

**Comment by carloscuesta** [comment]
Fixed the case found on snapshots with your suggestion @kacieb 😊

**Comment by amarlette** [comment]
Thank you @carloscuesta for this contribution to a more accessible React Native! I would like to give you a shout-out on Twitter and include you in our end-of-month issues update for your contribution. Do you have a Twitter we can tag?

**Comment by carloscuesta** [comment]
Sure @amarlette!

Here's my twitter handle [@crloscuesta](https://twitter.com/crloscuesta)

**Pull request closed without merging**